src/shared/ui/list/list.tsx

```diff
@@ -0,0 +1,44 @@
+// todo delete TextWithLink
+import classNames from 'classnames/bind';
+import { TextWithLink } from '../text-with-link';
+import { LinkList } from '@/widgets/required/required.types';
+
+import styles from './list.module.scss';
+
+export type ListProps = {
+  data: ListData;
+  marked?: boolean;
+};
+
+export type ListData = (string | LinkList)[] | [] | undefined;
+
+export const cx = classNames.bind(styles);
+
+export const List = ({ data, marked = true }: ListProps) => {
+  if (!data?.length) {
+    return <ul data-testid="list"></ul>;
+  }
```

Comment:
I don't see much sense in this condition, since removing it won't change anything

Reply:
I changed it to Fragment to return the HTMLElement anyway

Reply:
Why not just remove that condition?

Reply:
I suppose we can have some lists WO `data`

Reply:
> I suppose we can have some lists WO `data`

i think you can just change in main return to optional chaining, the result will be the same - empty ul if no data passed.

Reply:
We can have two variants of an incorrect `date` - it is `undefined` or it is an empty array. As far as I understood you, you suggest adding two conditions - to the `<ul> and to the data.map? It seems to me that my code is cleaner. Moreover, when we normalize the data structure, this condition can be simply removed.

Reply:
you simply need to remove

```typescript
 if (!data?.length) {
    return <></>;
 }
```


and add 

``` typescript
data?.map(
```

Reply:
Do we're really would like redundant tag `<ul>` in our code?